xwiki-platform-core/xwiki-platform-menu/xwiki-platform-menu-ui/src/main/resources/Menu/MenuMacro.xml

```diff
@@ -154,28 +154,79 @@ $menuTemplateDoc.content.replace('{', '')
       <cache>long</cache>
     </property>
     <property>
-      <code>require(['jquery'], function($) {
+      <code>define('menu-ui-translation-keys', {
+  prefix: 'menu.ui.',
+  keys: [
+    "openSubMenu",
+    "closeSubMenu"
+  ]
+});
+require(['jquery'], function($) {
   // It's not possible to write a CSS selector that targets list items containing lists so we rely on JavaScript.
   // The 'dropdown' CSS class is used only to display the down/left arrow.
-  $('.menu-horizontal li ul').parent().addClass('xDropdown');
+  $('.menu-horizontal li ul, .menu-vertical li ul').parent().addClass('xDropdown');
 
   // Make sure the menu separators are really empty.
   $('.menu-horizontal, .menu-vertical').find('li &gt; br:first-child').remove();
 
-  // Collapsible menu bahavior.
-  $('.menu-vertical.collapsible').each(function(){
+  // Add aria attributes to the different elements.
+  $('.menu-horizontal').attr('role', 'menubar');
+  $('.menu-vertical').attr('role', 'menu');
+  $('.xDropdown').each(function() {
+    $(this).attr('role', 'menuitem');
+    $(this).attr('aria-haspopup', 'menu');
+  });
+  $('.menu-horizontal, .menu-vertical').find('li')
+    .filter(function(index) {return this.textContent.trim() === ""; })
+    .attr("role","separator");
+
+  $('.menu-vertical.collapsible').each(function() {
     var open = $(this).hasClass('open');
     $(this).find('li ul').each(function() {
       $(this).addClass('xDropdown-menu').parent().addClass('xDropdown' + (open ? ' open' : ''));
-      // Wrap everything (including text nodes) before the sub-menu in a DIV that will toggle its state.
-      var toggle = this.ownerDocument.createElement('div');
-      $(this).parent().prepend(toggle);
-      for(var next = toggle.nextSibling; next != this; next = toggle.nextSibling) {
-        toggle.appendChild(next);
+    });
+  });
+
+  function setDropdownButtonTitle(dropDownButton) {
+    var xDropdown = $(dropDownButton).parent().parent();
+    require(['xwiki-l10n!menu-ui-translation-keys'], function(l10n) {
```

Comment:
This require can probably be moved to the 'main' require call l164.

Reply:
We are using `requirejs`. The doc can be found at https://requirejs.org/docs/api.html

Reply:
And the main endpoint for doc regarding XWiki frontend resources https://www.xwiki.org/xwiki/bin/view/Documentation/DevGuide/FrontendResources/JavaScriptAPI/

Reply:
Thanks! :)